### PR TITLE
Closes #4535: Integrate setEnhancedTrackingProtectionLevel geckoView api.

### LINKED ISSUE #4535: Integrate setEnhancedTrackingProtectionLevel  new GeckoView api
Bugzilla bug https://bugzilla.mozilla.org/show_bug.cgi?id=1580854

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt [hunk 1]
```diff
@@ -210,6 +210,51 @@ class GeckoEngineTest {
         verify(mockRuntime.settings.contentBlocking).setStrictSocialTrackingProtection(true)
     }
 
+        @Test
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: nit: formatting

File: components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt [hunk 2]
```diff
@@ -210,6 +210,51 @@ class GeckoEngineTest {
...
+
+        engine.settings.trackingProtectionPolicy = TrackingProtectionPolicy.recommended()
+
+        verify(mockRuntime.settings.contentBlocking).setEnhancedTrackingProtectionLevel(
+            ContentBlocking.EtpLevel.DEFAULT
+        )
+    }
+
+    @Test
+    fun `WHEN a tracking protection policy different recommended or strict from is set to recommended THEN the setEnhancedTrackingProtectionLevel must be NONE`() {
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineTest.kt]: nit: I think this test name needs to be rephrased? I would have just combined this all in one test with comments as you're just testing the different combinations...if you don't want that then how about something like 'WHEN a tracking protection policy other than recommended or strict from is set THEN the setEnhancedTrackingProtectionLevel must be NONE'.

**Comment by Amejia481** [comment]
I update the PR including the recommendations
bors r=csadilek

**Comment by pocmo** [comment]
bors rety

**Comment by pocmo** [comment]
bors retry

**Pull request merged**